### DISCUSSION TITLE
`fyne serve` default target

### DISCUSSION THREAD

**Discussion by srackham** [fyne-io/fyne#4480] [General]
The default `fyne serve` command target is "web" which requires the `gopherjs` compiler, which in turn only supports up to Go 1.18, so unless you use an old version of Go you get an error like:

```
$ fyne serve
GopherJS 1.17.2+go1.17.9 requires a Go 1.17.x distribution, but found version go1.21.5
time 2023-11-29T21:21:46Z
```

Would it make more sense to have `wasm` as the default target, or am I missing something?

Now that Go has WASM compilation support, does it make any sense to have a "gopherjs" (JavaScript) target?

**Comment by andydotxyz** [comment]
You don't have to use Go 1.18, you can use any as long as the older install is also there, see: https://developer.fyne.io/started/webapp.

There is an ongoing discussion, see #4330, it was considered on our latest contributor call that perhaps GopherJS should not be included going forward. It is being discussed.

However in response to the actual question - no, the technology should not be the name of a OS target as that's an implementation detail. A relevant question could be "should GopherJS be disabled by default" which makes this a duplicate question.

**Reply by srackham** [reply]
Brilliant! Thanks for  the super-fast response. As I work through the Fysion videos I continue to be amazed, until I stumbled on Fyne I had given up on cross-platform GUI development.

**Reply by andydotxyz** [reply]
I'm very glad this is exciting - so much is possible but there is a lot more to come in the year ahead. :)
